util/queue/queue.go

```diff
@@ -214,12 +214,10 @@ func (q *Queue) sleep(d time.Duration) {
 		log.Debug("Sleeping...")
 	}
 	sleep := time.NewTimer(d)
+	defer sleep.Stop()
 	start := time.Now()
 	select {
 	case <-q.signal:
-		if !sleep.Stop() {
-			<-sleep.C
```

Comment:
Good catch. Based on the documentation for `Timer.Stop` this will block forever if the timer already fired.
> It returns ... false if the timer has already expired or been stopped.

This seems like a pretty rare race since the select statement also watches the timer, but if `q.sleep()` is called when there is already a signal pending this would happen reliably.

Reply:
Yeah, the logs are not logging every occurence of this since they stick to INFO level, but I suspect given the frequency of updates that sleep and rouse are both happening quite frequently in production. (I'm guessing steady increase in traffic has made this more likely over time, so we're seeing it more often recently).